agent-core/src/main/java/com/newrelic/agent/android/harvest/Harvester.java

```diff
@@ -702,6 +725,27 @@ private void fireOnHarvestConnected() {
         }
     }
 
+    public void checkOfflineAndPersist() {
```

Comment:
Other than the feature flag, where does this method check offline status? Adding that here could remove a lot of duplicate checks leading up to this method.

Reply:
This is the wrap-up function used only within the harvester when response is unknown or response has network error.